components/util/remutex.rs

```diff
@@ -0,0 +1,217 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
No, it's the same interface as the libstd version, but implemented on top of std, rather than using reentrant locks from the OS.